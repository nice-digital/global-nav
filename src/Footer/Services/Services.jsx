import React, { Component } from "react";
import PropTypes from "prop-types";
import { createMarkup } from "../../utils";

import styles from "./Services.module.scss";
import TrackedLink from "../../TrackedLink";
import { footerClickEventAction } from "../../tracker";
import links from "../../services.json";

export default class Services extends Component {
	render() {
		return (
			<nav className={styles.wrapper} aria-label="Our services">
				<ul className={styles.list}>
					{links.external.map(
						function (link) {
							if (link.nestedLinks) {
								return link.nestedLinks.map((nestedLink) => {
									return <FooterLink key={nestedLink.id} {...nestedLink} />;
								});
							} else {
								return <FooterLink key={link.id} {...link} />;
							}
						}.bind(this)
					)}
				</ul>
			</nav>
		);
	}
}

FooterLink.propTypes = {
	href: PropTypes.string,
	id: PropTypes.string,
	text: PropTypes.string,
	abbreviation: PropTypes.bool,
	title: PropTypes.string,
};

function FooterLink({ href, text, abbreviation, title }) {
	return (
		<li>
			<TrackedLink
				href={href}
				eventAction={footerClickEventAction}
				eventLabel={text}
				className={styles.link}
			>
				{abbreviation ? (
					<>
						<abbr title={title}>
							{text} <span className={styles.visuallyHidden}>{title}</span>
						</abbr>
						<span aria-hidden="true" className={styles.tooltip}>
							{title}
						</span>
					</>
				) : (
					<>
						<span dangerouslySetInnerHTML={createMarkup(text)}></span>
					</>
				)}
			</TrackedLink>
		</li>
	);
}

Services.propTypes = {
	service: PropTypes.string,
};
