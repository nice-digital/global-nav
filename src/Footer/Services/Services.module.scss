@use 'sass:math';
@use '../../styles/global-nav-colours' as *;
@use '@nice-digital/nds-core/scss/media-queries';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/utils';
@use '../../Common.module.scss';

.wrapper {
  margin: utils.rem(0 0 spacing.$large);
}

.list {
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    margin: utils.rem(0 0 spacing.$x-small);
  }
}

@include media-queries.mq($from: sm) {
  .wrapper {
    border-right: 2px solid $services-border;
    float: left;
    padding-right: utils.rem(spacing.$medium);
    width: math.percentage(math.div(1, 3));
  }
}

@include media-queries.mq($from: md) {
  .wrapper {
    border-right: 2px solid $services-border;
    float: left;
    margin: 0;
    width: 25%;
  }
}
