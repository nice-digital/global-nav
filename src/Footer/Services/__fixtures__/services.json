{
  "external": [
    {
      "host": "www.test.nice.org",
      "landingPath": "/url1/",
      "text": "First link",
      "id": "link1",
      "footer": true
    },
    {
      "host": "www.test.nice.org",
      "landingPath": "/url2/",
      "text": "Second link",
      "id": "link2",
      "footer": false
    },
    {
      "host": "www.test.nice.org",
      "landingPath": "/url3/",
      "text": "Third link",
      "id": "link3",
      "footer": true
    }
  ]
}
