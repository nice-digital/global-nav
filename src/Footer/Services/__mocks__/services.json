{
  "external": [
    {
      "landingUrl": "https://url1/",
      "text": "First link",
      "id": "link1",
      "footer": true
    },
    {
      "landingUrl": "https://url2/",
      "text": "Second link",
      "id": "link2",
      "footer": false
    },
    {
      "landingUrl": "https://url3/",
      "text": "Third link",
      "id": "link3",
      "footer": true
    }
  ]
}
