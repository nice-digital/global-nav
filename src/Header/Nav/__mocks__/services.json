{
  "rootUrl": "https://rootUrl.org.uk",
  "external": [
    {
      "href": "https://url1/",
      "text": "First link",
      "id": "link1",
      "dropdown": true,
      "dropdownComponent": "FirstComponent"
    },
    {
      "href": "https://url2/",
      "text": "Second link",
      "id": "link2",
      "dropdown": false,
      "abbreviation": true,
      "title": "Abbreviation title",
      "links": [
        {
          "text": "Sub link 1",
          "href": "/"
        },
        {
          "text": "Sub link 2",
          "href": "/test"
        }
      ]
    },
    {
      "href": "https://url3/",
      "text": "Third link",
      "id": "link3",
      "dropdown": true,
      "dropdownComponent": "ThirdComponent",
      "abbreviation": true,
      "title": "Abbreviation title",
      "links": [
        {
          "text": "Sub link 1",
          "href": "/"
        },
        {
          "text": "Sub link 2",
          "href": "/test"
        }
      ],
      "nestedLinks": [
        {
          "href": "https://nestedLink1",
          "text": "Nested link one",
          "id": "nested-1"
        },
        {
          "href": "https://nestedLink2",
          "text": "Nested link two",
          "id": "nested-2"
        }
      ]
    }
  ],
  "internal": [
    {
      "href": "/",
      "text": "First internal service",
      "id": "internal1",
      "links": [
        {
          "text": "internal1 sub link 1",
          "href": "/i1sublink1"
        },
        {
          "text": "internal1 sub link 2",
          "href": "/i2sublink2"
        }
      ]
    },
    {
      "href": "/",
      "text": "Second internal service",
      "id": "internal2",
      "links": [
        {
          "text": "internal2 sub link 1",
          "href": "/i2sublink1"
        },
        {
          "text": "internal2 sub link 2",
          "href": "/i2sublink2"
        }
      ]
    }
  ]
}
