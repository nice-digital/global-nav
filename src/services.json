{
  "external": [
    {
      "href": "https://pathways.nice.org.uk/",
      "text": "NICE Pathways",
      "id": "pathways",
      "links": [
        {
          "text": "Browse",
          "href": "/#browse"
        },
        {
          "text": "Recently viewed",
          "href": "/#recently-viewed"
        },
        {
          "text": "Help",
          "href": "/#help"
        },
        {
          "text": "Leave feedback",
          "href": "https://www.nice.org.uk/NICE-Pathways-feedback"
        }
      ]
    },
    {
      "href": "https://www.nice.org.uk/guidance",
      "text": "NICE guidance",
      "id": "guidance"
    },
    {
      "href": "https://www.nice.org.uk/about/what-we-do/life-sciences",
      "text": "Life sciences",
      "id": "life-sciences"
    },
    {
      "href": "https://www.nice.org.uk/standards-and-indicators",
      "text": "Standards and indicators",
      "id": "standards"
    },
    {
      "href": "https://www.evidence.nhs.uk/",
      "text": "Evidence search",
      "id": "evidence"
    },
    {
      "href": "https://bnf.nice.org.uk/",
      "text": "BNF",
      "abbreviation": true,
      "title": "British National Formulary",
      "id": "bnf",
      "links": [
        {
          "text": "Drugs",
          "href": "/drug/"
        },
        {
          "text": "Interactions",
          "href": "/interaction/"
        },
        {
          "text": "Treatment summaries",
          "href": "/treatment-summary/"
        },
        {
          "text": "What's changed?",
          "href": "/about/changes.html"
        },
        {
          "text": "About BNF",
          "href": "/about/"
        }
      ]
    },
    {
      "href": "https://bnfc.nice.org.uk/",
      "text": "BNFC",
      "abbreviation": true,
      "title": "British National Formulary for Children",
      "id": "bnfc",
      "links": [
        {
          "text": "Drugs",
          "href": "/drug/"
        },
        {
          "text": "Interactions",
          "href": "/interaction/"
        },
        {
          "text": "Treatment summaries",
          "href": "/treatment-summary/"
        },
        {
          "text": "What's changed?",
          "href": "/about/changes.html"
        },
        {
          "text": "About BNFC",
          "href": "/about/"
        }
      ]
    },
    {
      "href": "https://cks.nice.org.uk/",
      "text": "CKS",
      "abbreviation": true,
      "title": "Clinical Knowledge Summaries",
      "id": "cks",
      "links": [
        {
          "image": "cks",
          "text": "CKS - Home",
          "href": "/"
        },
        {
          "text": "Health topics A to Z",
          "href": "/topics/"
        },
        {
          "text": "Specialities",
          "href": "/specialities/"
        },
        {
          "text": "What's new",
          "href": "/whats-new/"
        },
        {
          "text": "About CKS",
          "href": "/about/"
        }
      ]
    },
    {
      "href": "https://www.nice.org.uk/about/what-we-do/evidence-services/journals-and-databases",
      "text": "Journals and databases",
      "id": "journals"
    }
  ],
  "internal": [
    {
      "href": "/",
      "text": "NICE In development",
      "id": "indev",
      "links": [
        {
          "text": "Guidance projects",
          "href": "/"
        },
        {
          "text": "Advice projects",
          "href": "/adviceprojects"
        }
      ]
    }
  ]
}
