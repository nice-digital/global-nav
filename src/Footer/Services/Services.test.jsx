import React from "react";

import Services from "./Services";

import { render, within } from "@testing-library/react";

import services from "./__mocks__/services.json";

jest.mock("./../../services.json", () => require("./__mocks__/services.json"));

describe("Services", () => {
	it("Matches snapshot", () => {
		const { container } = render(<Services service="pathways" />);
		expect(container).toMatchSnapshot();
	});

	it("should render link(s) if footer boolean is set to true", () => {
		const { getByRole } = render(<Services service="cks" />);

		const servicesNav = getByRole("navigation", { name: "Our services" });

		const servicesLinks = within(servicesNav).getAllByRole("link");

		const expectedServiceLabels = services.external
			.filter((s) => s.footer)
			.map((s) => s.text);

		expect(servicesLinks.map((l) => l.textContent)).toStrictEqual(
			expectedServiceLabels
		);
	});
});
