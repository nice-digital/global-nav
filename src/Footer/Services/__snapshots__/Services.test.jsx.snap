// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Matches snapshot 1`] = `
<nav
  aria-label="Our services"
  className="wrapper"
>
  <ul
    className="list"
  >
    <li
      key="pathways"
    >
      <TrackedLink
        aria-current="true"
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="NICE Pathways"
        href="https://pathways.nice.org.uk/"
      >
        NICE Pathways
      </TrackedLink>
    </li>
    <li
      key="guidance"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="NICE guidance"
        href="https://www.nice.org.uk/guidance"
      >
        NICE guidance
      </TrackedLink>
    </li>
    <li
      key="life-sciences"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Life sciences"
        href="https://www.nice.org.uk/about/what-we-do/life-sciences"
      >
        Life sciences
      </TrackedLink>
    </li>
    <li
      key="standards"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Standards and indicators"
        href="https://www.nice.org.uk/standards-and-indicators"
      >
        Standards and indicators
      </TrackedLink>
    </li>
    <li
      key="evidence"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Evidence search"
        href="https://www.evidence.nhs.uk/"
      >
        Evidence search
      </TrackedLink>
    </li>
    <li
      key="bnf"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="BNF"
        href="https://bnf.nice.org.uk/"
      >
        <abbr
          title="British National Formulary"
        >
          BNF
           
          <span
            className="visuallyHidden"
          >
            British National Formulary
          </span>
        </abbr>
        <span
          aria-hidden="true"
          className="tooltip"
        >
          British National Formulary
        </span>
      </TrackedLink>
    </li>
    <li
      key="bnfc"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="BNFC"
        href="https://bnfc.nice.org.uk/"
      >
        <abbr
          title="British National Formulary for Children"
        >
          BNFC
           
          <span
            className="visuallyHidden"
          >
            British National Formulary for Children
          </span>
        </abbr>
        <span
          aria-hidden="true"
          className="tooltip"
        >
          British National Formulary for Children
        </span>
      </TrackedLink>
    </li>
    <li
      key="cks"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="CKS"
        href="https://cks.nice.org.uk/"
      >
        <abbr
          title="Clinical Knowledge Summaries"
        >
          CKS
           
          <span
            className="visuallyHidden"
          >
            Clinical Knowledge Summaries
          </span>
        </abbr>
        <span
          aria-hidden="true"
          className="tooltip"
        >
          Clinical Knowledge Summaries
        </span>
      </TrackedLink>
    </li>
    <li
      key="journals"
    >
      <TrackedLink
        aria-current={null}
        className="link"
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Journals and databases"
        href="https://www.nice.org.uk/about/what-we-do/evidence-services/journals-and-databases"
      >
        Journals and databases
      </TrackedLink>
    </li>
  </ul>
</nav>
`;
